application/src/test/java/org/opentripplanner/street/integration/WalkRoutingTest.java

```diff
@@ -59,15 +59,33 @@ void pathReversalWorks(int offset) {
     var end = GenericLocation.fromCoordinate(59.94641, 10.77522);
     var base = dateTime.truncatedTo(ChronoUnit.SECONDS);
     var time = base.plusMillis(offset);
-    var results = route(roundabout, start, end, time, true);
-    assertEquals(1, results.size());
-    var states = results.get(0).states;
-    var diff = ChronoUnit.MILLIS.between(
-      states.getFirst().getTimeAccurate(),
-      states.getLast().getTimeAccurate()
+
+    var baseResults = route(roundabout, start, end, base, false);
+    assertEquals(1, baseResults.size());
+    var baseStates = baseResults.get(0).states;
+    var baseDiff = ChronoUnit.MILLIS.between(
+      baseStates.getFirst().getTimeAccurate(),
+      baseStates.getLast().getTimeAccurate()
+    );
+
+    var departFromResults = route(roundabout, start, end, time, false);
+    assertEquals(1, departFromResults.size());
+    var departAtStates = departFromResults.get(0).states;
+    var departFromDiff = ChronoUnit.MILLIS.between(
+      departAtStates.getFirst().getTimeAccurate(),
+      departAtStates.getLast().getTimeAccurate()
+    );
+
+    var arriveByResults = route(roundabout, start, end, time, true);
+    assertEquals(1, arriveByResults.size());
+    var arriveByStates = arriveByResults.get(0).states;
+    var arriveByDiff = ChronoUnit.MILLIS.between(
+      arriveByStates.getFirst().getTimeAccurate(),
+      arriveByStates.getLast().getTimeAccurate()
     );
-    // should be same for every parametrized offset, otherwise irrelevant
-    assertEquals(13926, diff);
+    // should be same for every parametrized offset and both depart at and arrive by
+    assertEquals(baseDiff, departFromDiff);
+    assertEquals(baseDiff, arriveByDiff);
```

Comment:
What is the reasoning behind this change?

Reply:
The number is meaningless. The purpose of this test is to ensure that the same number is returned in every case. This ensure that future changes in the routing algorithm won't invalidate this test.

Reply:
I think there is a good reason to keep the number as a constant. Tests should have a reliable outcome. This can serve as a canary to see if some change has a drastic effect. Lets discuss this in the next dev meeting